src/azure-cli/azure/cli/command_modules/containerapp/_compose_utils.py

```diff
@@ -0,0 +1,373 @@
+    unsupported_configuration = ["volumes", "volumes_from"]
+    config_list = resolve_configuration_element_list(compose_service, unsupported_configuration)
+    message = "These volume mount elements from the docker-compose file are not supported"
+    message += " in Azure Container Apps. For more information about supported storage configuration,"
+    message += " please see https://aka.ms/containerapp/compose/volumes"
+    if len(config_list) >= 1:
+        logger.warning(message)
+        for item in config_list:
+            logger.warning("     %s", item)
+
+
+def warn_about_unsupported_network(compose_service):
+    unsupported_configuration = ["networks", "network_mode", "mac_address"]
+    config_list = resolve_configuration_element_list(compose_service, unsupported_configuration)
+    message = "These network configuration settings from the docker-compose file are not supported"
+    message += " in Azure Container Apps. For more information about supported networking configuration,"
+    message += " please see https://aka.ms/containerapp/compose/networking"
+    if len(config_list) >= 1:
+        logger.warning(message)
+        for item in config_list:
+            logger.warning("     %s", item)
+
+
+def warn_about_unsupported_elements(compose_service):
+    warn_about_unsupported_build_configuration(compose_service)
+    warn_about_unsupported_runtime_host_configuration(compose_service)
+    warn_about_unsupported_volumes(compose_service)
+    warn_about_unsupported_network(compose_service)
+
+
+def check_supported_platform(platform):
+    if platform is not None:
+        platform = platform.split('/')
+        if len(platform) >= 2:
+            return platform[0] == 'linux' and platform[1] == 'amd64'
+        return platform[0] == 'linux'
+    return True
+
+
+def service_deploy_exists(service):
+    return service.deploy is not None
+
+
+def service_deploy_resources_exists(service):
+    return service_deploy_exists(service) and service.deploy.resources is not None
+
+
+def flatten_list(source_value):
+    flat_list = []
+    for sub_list in source_value:
+        flat_list += sub_list
+    return flat_list
+
+
+def resolve_transport_from_cli_args(service_name, transport):
+    if transport is not None:
+        transport = flatten_list(transport)
+        for setting in transport:
+            key, value = setting.split('=')
+            if key.lower() == service_name.lower():
+                return value
+    return 'auto'
+
+
+def resolve_registry_from_cli_args(registry_server, registry_user, registry_pass):
+    if registry_server is not None:
+        if registry_user is None and registry_pass is None:
+            registry_user = prompt("Please enter the registry's username: ")
+            registry_pass = prompt("Please enter the registry's password: ")
+        elif registry_user is not None and registry_pass is None:
+            registry_pass = prompt("Please enter the registry's password: ")
```

Comment:
Is it possible that `registry_pass` is input without `registry_user`?

Reply:
Yes. Currently if `registry_user `or `registry_pass `is None, it will try to fetch creds. So if `registry_pass `is input without `registry_user`, the `registry_pass `will not be handled.

```
 if app.registry_user is None or app.registry_pass is None:
            registry_rg = _set_acr_creds(cmd, app, registry_name)
```